### ISSUE TITLE
VkDescriptorSetLayoutBinding.stageFlags is of type uint

### ISSUE THREAD

**Issue by HurricanKai** [terrafx/terrafx.interop.vulkan#15]
#### Description (optional)
I'm not sure if this should be a Bug, or as what this should be filed, or if it's wrong at all, but I wanted to note it

#### Expected Behavior

VkDescriptorSetLayoutBinding.stageFlags is of type `VkShaderStageFlagBits`

#### Actual Behavior

VkDescriptorSetLayoutBinding.stageFlags is of type `uint`

**Comment by tannergooding** [comment]
So the reason I don't remap things like this is for a few reasons:
* It differs from the actual completion/etc that C/C++ users get
* It is a source/binary breaking change for types if v1.0 uses `uint` and a later version uses `VkStageFlags` (hence why Vulkan just uses a typedef)

In general, except where there are multiple names where one is almost exclusively used (and therefore differing from a typedef might hurt the ability to port code), I've favored using the exact name as given in native code (some examples of where I've allowed a rewrite are: `HWND__` to `HWND`; `tagPOINT` to `POINT`, or `_MyStruct` to `MyStruct`).

In this case, C/C++ would require the user to either do `VkStageFlagsBits::Name` or, more commonly, people just use `Name` directly. They would never use `VkStageFlags` except in their own signatures (and that would ultimately be typed down to `uint` as well).
